internal/backend/runtime/omni/controllers/omni/cluster_machine_config_status.go

```diff
@@ -689,6 +704,18 @@ func (h *clusterMachineConfigStatusControllerHandler) reset(
 		graceful = false
 	}
 
+	nodeForceDestroyRequest, err := safe.ReaderGetByID[*omni.NodeForceDestroyRequest](ctx, h.r, machineConfig.Metadata().ID())
+	if err != nil && !state.IsNotFoundError(err) {
+		return fmt.Errorf("failed to get node force delete request %q: %w", machineConfig.Metadata().ID(), err)
+	}
+
+	forceDestroy := nodeForceDestroyRequest != nil
+	if forceDestroy {
+		logger.Info("node is requested to be force destroyed")
+
+		graceful = false
+	}
```

Comment:
Maybe lets extract that to a separate method?

Reply:
good idea, done